sdks/fabric/interoperation-node-sdk/src/InteroperableHelper.ts

```diff
@@ -170,19 +171,64 @@ const verifyRemoteProposalResponse = async (proposalResponseBase64, isEncrypted,
 const getResponseDataFromView = (view, privKeyPEM = null) => {
     if (view.getMeta().getProtocol() == statePb.Meta.Protocol.FABRIC) {
         const fabricView = fabricViewPb.FabricView.deserializeBinary(view.getData());
-        const interopPayload = interopPayloadPb.InteropPayload.deserializeBinary(Uint8Array.from(Buffer.from(fabricView.getResponse().getPayload())));
-        if (interopPayload.getConfidential()) {    // Currently this is only supported for Fabric because it uses ECDSA keys in wallets
-            const confidentialPayload = interopPayloadPb.ConfidentialPayload.deserializeBinary(Uint8Array.from(Buffer.from(interopPayload.getPayload())));
-            const decryptedPayload = decryptData(Buffer.from(confidentialPayload.getEncryptedPayload()), privKeyPEM);
-            const decryptedPayloadContents = interopPayloadPb.ConfidentialPayloadContents.deserializeBinary(Uint8Array.from(Buffer.from(decryptedPayload)));
-            return { interopPayload: interopPayload, data: Buffer.from(decryptedPayloadContents.getPayload()).toString(), contents: decryptedPayload };
+        const fabricViewProposalResponses = fabricView.getEndorsedProposalResponsesList();
+        let viewAddress = '';
+        let responsePayload = '';
+        let responsePayloadContents = null;
+        let paymentConfidential = false;
```

Comment:
What does `paymentConfidential` name signify?

Reply:
It's a flag to match the confidentiality status (i.e., whether plaintext or encrypted) of all the proposal response payloads in the list. It helps us catch weird errors where one payload has the confidentiality flag on and another off.

Reply:
okay, I mean what does `payment` signify?

Reply:
Did you mean `payloadConfidential`?

Reply:
Yes, that was a typo. Good catch!